src/parsers.rs

```diff
@@ -169,12 +181,26 @@ impl<'a> Parser<'a> {
                 // Valid u16 to be added to the address
                 head[*head_idx] = part as u16;
                 *head_idx += 1;
+
+                if part == 0xffff && *head_idx == 6 {
+                    self.try(|p| {
+                        p.accept_char(b':')?;
+                        p.accept_ipv4_mapped_ipv6_part(head, head_idx)
+                    });
+                }
                 Ok(())
             },
             Some(part) if *tail_idx < 6 => {
                 // Valid u16 to be added to the address
                 tail[*tail_idx] = part as u16;
                 *tail_idx += 1;
+
+                if part == 0xffff && use_tail && *tail_idx == 1 {
```

Comment:
AFAIK `use_tail` should be always true in this branch.

Reply:
yes, removed